core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -697,6 +697,9 @@ wasm_cluster_join_thread(WASMExecEnv *exec_env, void **ret_val)
 int32
 wasm_cluster_detach_thread(WASMExecEnv *exec_env)
 {
+    if (exec_env->thread_is_detached)
+        return 0;
```

Comment:
Should move the code into the lock/unlock code piece, e.g. :
```C
    os_mutex_lock(&cluster_list_lock);
    if (exec_env->thread_is_detached) {
        os_mutex_unlock(&cluster_list_lock);
        return 0;
    }
    if (!clusters_have_exec_env(exec_env)) {
    ...
    os_mutex_unlock(&cluster_list_lock);
```
Or if another thread changes the state when current thread runs between L701 and L705, current thread will change again.

Reply:
OK, I have moved it into lock protection.